addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -109,13 +109,21 @@ def _import_fattura_pa(self, tree, invoice):
                 invoice = self.env['account.move']
             first_run = False
 
+            # Refund type.
+            # TD01 == invoice
+            # TD02 == advance/down payment on invoice
+            # TD03 == advance/down payment on fee
+            # TD04 == credit note
+            # TD05 == debit note
+            # TD06 == fee
+            # For unsupported document types, just assume in_invoice, and log that the type is unsupported
             elements = tree.xpath('//DatiGeneraliDocumento/TipoDocumento')
-            if elements and elements[0].text and elements[0].text == 'TD01':
-                self_ctx = invoice.with_context(default_move_type='in_invoice')
-            elif elements and elements[0].text and elements[0].text == 'TD04':
-                self_ctx = invoice.with_context(default_move_type='in_refund')
-            else:
-                _logger.info('Document type not managed: %s.', elements[0].text)
+            move_type = "in_invoice"
+            if elements and elements[0].text and elements[0].text == 'TD04':
+                move_type = "in_refund"
+            elif elements and elements[0].text and elements[0].text != 'TD01':
+                _logger.info('Document type not managed: %s. Invoice type is set by default.', elements[0].text)
+            self_ctx = invoice.with_context(default_move_type=move_type)
```

Comment:
It would be indeed clearer to name the variable `invoice_ctx` instead of `self_ctx`. Here, we are working with `invoice` really that is reset each iteration if there are multiple invoices in the xml.

Reply:
Ok